.circleci/config.yml

```diff
@@ -1,183 +1,60 @@
-    <<: *linux_defaults
+  checkout_code:
+    executor: rn/linux_js
     steps:
       - checkout
-      - restore-cache: *restore-yarn-cache
-      - run: rm -rf node_modules
-      - run: yarn cache clean
-      - run: *yarn
-      - save-cache: *save-yarn-cache
       - persist_to_workspace:
           root: .
           paths: .
-
-  eslint:
-    <<: *linux_defaults
-    steps:
-    - attach_workspace:
-        at: ~/react-native-viewpager
-    - run: *eslint
-
-  flow:
-    <<: *linux_defaults
+  analyse_js:
+    executor: rn/linux_js
     steps:
       - attach_workspace:
-          at: ~/react-native-viewpager
-      - run: *flow
-
-  # jest:
-  #   <<: *linux_defaults
-  #   steps:
-  #   - attach_workspace:
-  #       at: ~/react-native-viewpager
-  #   - run: *jest
-
-  android-compile:
-    <<: *android_defaults
-    steps:
-      - attach_workspace:
-          at: ~/react-native-viewpager
-      - restore-cache: *restore-gradle-cache
-      - run:
-          name: Accept Android licences
-          command: |-
-            yes | sdkmanager --licenses || exit 0
-            yes | sdkmanager --update || exit 0
+          at: .
+      - rn/yarn_install
       - run:
-          name: Build Android Example App and Library
-          command: |-
-            cd example/android
-            ./gradlew clean assembleDebug
-      - save-cache: *save-gradle-cache
-
-  ios-checkout:
-    <<: *macos_defaults
-    steps:
-      - checkout
-      - restore-cache: *restore-yarn-cache
-      - run: rm -rf node_modules
-      - run: yarn cache clean
-      - run: *yarn
-      - save-cache: *save-yarn-cache
-      - persist_to_workspace:
-          root: .
-          paths: .
-
-  ios-compile:
-    <<: *macos_defaults
-    steps:
-      - attach_workspace:
-          at: ~/react-native-viewpager
+          name: Run ESLint
+          command: yarn lint
       - run:
-          name: Build example app
-          command: |-
-            react-native run-ios --project-path example/ios
+          name: Flow
+          command: yarn flow
 
-# -------------------------
-#        WORKFLOWS
-# -------------------------
 workflows:
-  version: 2
-  Test:
+  test:
     jobs:
-      - linux-checkout
-      - eslint:
+      # Checkout the code and persist to the Workspace
+      # Note: This is a job which is defined above and not part of the Orb
+      - checkout_code
+
+      # Analyze the Javascript using ESLint, Flow, and Jest
+      # Note: This is a job which is defined above and not part of the Orb
+      - analyse_js:
           requires:
-          - linux-checkout
-      - flow:
+            - checkout_code
+
+      # Build the Android app in debug mode
+      - rn/android_build:
+          name: build_android
+          project_path: "example/android"
+          build_type: debug
           requires:
-            - linux-checkout
-      # - jest:
+            - analyse_js
+
+      # Build the iOS app in release mode and do not run tests
```

Comment:
Do you need it?

Reply:
I would like to check, if iOS building correctly, but probably, this repo has no permission to macOS container. Now I have no time to check it. I would like to do it later.